dimensiondata/src/main/java/org/jclouds/dimensiondata/cloudcontrol/domain/IdeController.java

```diff
@@ -18,42 +18,50 @@
 
 import com.google.auto.value.AutoValue;
 import com.google.common.collect.ImmutableList;
-import org.jclouds.javax.annotation.Nullable;
 import org.jclouds.json.SerializedNames;
 
 import java.util.List;
 
 @AutoValue
-public abstract class IdeController extends AbstractBaseController {
+public abstract class IdeController {
 
-   public abstract Integer channel();
+   public abstract String id();
 
-   @Nullable
-   public abstract List<AbstractDrive> deviceOrDisks();
```

Comment:
Oh, I see now that inheritance here was there not only to reuse fields but because the API returns collections with different object types in it.
To avoid having a `List<Object>`, which makes the API hard to consume, would it make sense to have just one object with all fields? I see the child classes just had one or two different fields. Would that make sense?
If not, I'd consider getting the abstract classes back and just be extra-careful in variable order. take into account that the fields in the base class are serialized first, then the ones in the child class (and if at some point the base class is modified, the annotations in all child classes need to be updated accordingly).
Apologies for the noise if you consider the second option to be better, but till now it wasn't clear for me the purpose of the abstract classes.

Reply:
Yes, AbstractDrive will help a lot in ServerToHardware function implementation for a new ServerApi version. Gonna keep it then.